jenkins/data-prepper/distribution-artifacts.jenkinsfile

```diff
@@ -0,0 +1,80 @@
+lib = library(identifier: 'jenkins@20211123', retriever: legacySCM(scm))
+
+pipeline {
+    agent {
+        docker {
+            label 'Jenkins-Agent-al2-x64-c54xlarge-Docker-Host'
+            image 'opensearchstaging/ci-runner:ci-runner-centos7-v1'
+            alwaysPull true
+        }
+    }
+    parameters {
+        string(name: 'VERSION', description: 'The version of Data Prepper', trim: true)
+        string(name: 'BRANCH', description: 'Can be: <branchName>, <commitId>, refs/tags/<tagName>', trim: true)
+    }
+    stages {
+        stage('Build Data Prepper') {
+            steps {
+                script {
+                    checkout changelog: false, poll: false, scm:
+                            [$class: 'GitSCM', branches: [[name: BRANCH]], userRemoteConfigs: [[url: 'https://github.com/opensearch-project/data-prepper.git']]]
+                    sh './gradlew clean :release:archives:linux:linuxTar -Prelease'
+                    artifactsPath = "$WORKSPACE/release/archives/linux/build/distributions"
+                    artifactFileNames = sh(script: "find ${artifactsPath} | sed -n \"s|^${artifactsPath}/||p\"", returnStdout: true).split()
+                }
+            }
+        }
+        stage('Sign Artifacts') {
+            steps {
+                script {
+                    signatureType = '.sig'
+                    signArtifacts(
+                            artifactPath: "${artifactsPath}",
+                            signatureType: signatureType,
+                            distributionPlatform: 'linux'
+                    )
+                }
+            }
+        }
+        stage('Upload Artifacts to Staging Distribution Bucket') {
+            steps {
+                script {
+                    uploadPath = ([
+                            "${JOB_NAME}",
+                            "${VERSION}",
+                            "${BUILD_NUMBER}",
+                            "builds",
+                            "signed"
+                    ].join('/'))
+
+                    uploadToS3(
+                            sourcePath: "${artifactsPath}",
+                            bucket: "${ARTIFACT_BUCKET_NAME}",
+                            path: uploadPath
+                    )
+                    printUrls(artifactFileNames)
+                }
+            }
+        }
+    }
+    post() {
+        always {
+            script {
+                postCleanup()
+            }
+        }
+    }
+}
+
+void printUrls(artifactFileNames){
```

Comment:
I think @abhinavGupta16 is adding a helper for this elsewhere, let's use that.

Reply:
@abhinavGupta16 did create and add this helper. I'm unsure if he is adding it in general or not.

Reply:
I realized that this utility can be used by a lot of other jobs in the future and another currently. So I moved it `vars`. I have updated the job to accommodate that.

Reply:
Thank you @abhinavGupta16 !